plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java

```diff
@@ -151,4 +190,40 @@ public static HudiSplitWeightProvider createSplitWeightProvider(ConnectorSession
         }
         return HudiSplitWeightProvider.uniformStandardWeightProvider();
     }
+
+    static boolean partitionMatchesPredicate(
+            HudiSplit split,
+            TupleDomain<HiveColumnHandle> dynamicFilterPredicate)
```

Comment:
We should test partitioned tables with one or more partition fields, with all or subset of partition fields present in the dynamic filter predicates.

Reply:
See:

1. testDynamicFilterEnabledPredicatePushdown
2. testDynamicFilterDisabledPredicatePushdown

Reply:
The tests look good!